### PR TITLE
Show events with shifts on diff days on all days Closes #225

### LINKED ISSUE #225: For every day, show all events that have shifts on that day, that have not ended yet
Right now the events list contains every event only once, based on the date of its first shift beginning date that is still in the future.

In production this has caused two problems:

- [x] Shifts that have already started are hidden instantly (although the NGO might still be searching for volunteers during the shift)
- [ ] Volunteers often plan ahead a few days, e.g. they want to see where they can help on the next weekend. So they look at the date for the weekend but it's either empty or many events that have shifts on that date are not included on the list (because they are listed somewhere above on an earlier date).

Proposal:

1. **Hide events in the "Today" date group after they have ended**, instead of after they began.
2. **For every day, show all events with all shifts on that day** (i.e. group by date first, then group the shifts of event). So if one event has multiple shifts on the same day, it's listed once. But if it has shifts on two days, it's listed twice.

Ad 2: This will make the list much longer but I think it's worth it because the "Look up a date and see where I can help on that date" use case is so important. For performance we might have to look into pagination or caching (later), but I would keep it simple for now and see if we run into any problems. One thing we can do is **limit the list to dates in the next 4 weeks**.

### PR DESCRIPTION
**Pull request by pferdefleisch**
I'm not proud of this code but it is tested and it works. It is also
somewhat brittle and I think a few of the assumptions made are not sound

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -28,6 +28,9 @@ gem 'ri_cal'
 gem 'rails_autolink'
 gem 'faker', require: false
 
+gem 'pry-rails'
+gem 'pry-byebug', require: false
```
Comment (klappradla) [inline Gemfile]: Why did you not *fully* switch over to `pry`? You could have removed `byebug` in the test and dev and instead throw in `pry-byebug` there. Or do you see a usage for `byebug`there still?

File: app/models/event.rb [hunk 2]
```diff
@@ -20,9 +20,9 @@ class Event < ApplicationRecord
     order("shifts.starts_at").
     references(:available_shifts)
   }
-  scope :upcoming, -> { where(id: Shift.upcoming.pluck(:event_id).uniq) }
-  scope :past, -> { where(id: Shift.past.pluck(:event_id).uniq) }
-  scope :pending, -> { where(published_at: nil) }
+  scope :upcoming,  -> { where(id: Shift.upcoming.pluck(:event_id).uniq) }
```
Comment (klappradla) [inline app/models/event.rb]: 👍  I always hesitated to do it, felt to German doing it 🙈

File: app/models/shift.rb [hunk 3]
```diff
@@ -19,6 +19,35 @@ class Shift < ApplicationRecord
   before_destroy :notify_volunteers_about_destroy, prepend: true
   before_update  :notify_volunteers_about_update, prepend: true
 
+  def self.filtered_for_ngo(ngo, filters)
+    the_scope, order_by = filters
+    # unscoped is necessary because default_scope (deprecated?) is
+    # messing up the order by
+    shifts =
+      unscoped
```
Comment (klappradla) [inline app/models/shift.rb]: `unscoped` will include deleted shifts if not explicitly excluding them with e.g. `deleted_at not null`from what I remember.

We're using `paranoia` for soft deleting basically everything (insurance reasons) and therefore have a `default_scope` to neither include nor index marked as deleted records.

Comment (pferdefleisch) [inline app/models/shift.rb]: `default_scope` is stomping on my weird `order by`. I will have to add the null and deleted_at explicitly. Thanks for the heads up.

File: app/models/event.rb (hunk 2 above)
Comment (pferdefleisch) [inline app/models/event.rb]: I do this A LOT these days. I don't know what has come over me.

File: Gemfile (hunk 1 above)
Comment (pferdefleisch) [inline Gemfile]: I wasn't thinking about where I would use byebug. It isn't necessary outside of dev and test. I will update it.

File: app/models/shift.rb (hunk 3 above)
Comment (pferdefleisch) [inline app/models/shift.rb]: (but so is my code ;) )

Comment (pferdefleisch) [inline app/models/shift.rb]: fixed here
https://github.com/where2help/where2help/pull/238/commits/2f312374a425efd83f68645b88e0344247126e15

**Pull request merged**